encoders/firebase-encoders-processor/src/main/java/com/google/firebase/encoders/processor/Names.java

```diff
@@ -0,0 +1,45 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.firebase.encoders.processor;
+
+import javax.lang.model.element.Element;
+import javax.lang.model.element.PackageElement;
+
+public class Names {
+  /**
+   * Returns the class name that corresponds to a given element.
+   *
+   * <p>For top level classes returns the class name unchanged. For nested classes returns {@code
+   * ParentNested} for {@code Parent$Nested} classes.
+   */
+  static String generatedClassName(Element element) {
+    StringBuilder sb = new StringBuilder(element.getSimpleName().toString());
+    Element enclosingElement = element.getEnclosingElement();
+    while (!(enclosingElement instanceof PackageElement)) {
```

Comment:
Could you add a separator like AutoValue's `_`?

Reply:
I am not a fan of `_`, also fwiw dagger2 does not use them either, so unless you feel strongly I'd prefer not to have them.